src/gui/TimeLineWidget.cpp

```diff
@@ -339,20 +340,8 @@ void TimeLineWidget::mousePressEvent( QMouseEvent* event )
 		const TimePos t = m_begin + static_cast<int>( qMax( event->x() - m_xOffset - m_moveXOff, 0 ) * TimePos::ticksPerBar() / m_ppb );
 		const TimePos loopMid = ( m_loopPos[0] + m_loopPos[1] ) / 2;
 
-		if( t < loopMid )
-		{
-			m_action = MoveLoopBegin;
-		}
-		else if( t > loopMid )
-		{
-			m_action = MoveLoopEnd;
-		}
-
-		if( m_loopPos[0] > m_loopPos[1]  )
-		{
-			qSwap( m_loopPos[0], m_loopPos[1] );
-		}
-
+		m_action = t < loopMid ? MoveLoopBegin : MoveLoopEnd;
+		std::sort(m_loopPos, std::end(m_loopPos));
```

Comment:
I think `std::begin(m_loopPos)` would be better here - `m_loopPos` alone only works in this case because an array decays to a pointer to its first element, which happens to be the same type as `std::end(m_loopPos)`. Most iterable types aren't compatible with their iterator types (the only other ones that come to mind are the directory iterators in the filesystem library), so this wouldn't work if we changed from a raw C-style array to `std::array`, for instance.

Reply:
Good point! Fixed locally.